src/plugins/layout/nodeinfo.ts

```diff
@@ -27,7 +26,8 @@ export function shouldIgnoreNode(node: Node, parentIgnored: boolean): boolean {
         break;
       case Node.TEXT_NODE:
         // If we capture CSSRules on style elements, ignore its text children nodes
-        if (config.cssRules && node.parentNode && (node.parentNode as Element).tagName === "STYLE") {
+        let parent = node.parentNode as Element;
+        if (parent && parent.tagName === "STYLE" && parent.textContent.length === 0) {
```

Comment:
>tagName [](start = 29, length = 7)

What is this logic doing? add comment.

Reply:
Added some more comments although the comment above is still accurate.

---
In reply to: [243140382](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243140382) [](ancestors = 243140382)